cmd/agent/main.go

```diff
@@ -1,3 +1,43 @@
 package main
 
-func main() {}
+import (
+	"fmt"
+	"github.com/Kopleman/metcol/internal/agent/config"
+	"github.com/Kopleman/metcol/internal/agent/metrics-collector"
+	"github.com/Kopleman/metcol/internal/common/http-client"
+	"time"
+)
+
+func main() {
+	agentConfig := config.ParseAgentConfig()
+
+	endPointURL := `http://` + agentConfig.EndPoint.String() + `/update/`
+
+	httpClient := httpclient.NewHTTPClient(endPointURL)
+	collector := metricscollector.NewMetricsCollector(httpClient)
+	now := time.Now()
+
+	pollDuration := time.Duration(agentConfig.PollInterval) * time.Second
+	reportDuration := time.Duration(agentConfig.ReportInterval) * time.Second
+
+	collectTimer := now.Add(pollDuration)
+	reportTimer := now.Add(reportDuration)
+
+	for {
+		time.Sleep(1 * time.Second)
+
+		now = time.Now()
+		if now.After(collectTimer) {
+			collector.CollectMetrics()
+			fmt.Println(fmt.Printf("collected metrics at %s", time.Now().UTC()))
```

Comment:
1. Ты можешь написать так:

```suggestion
			fmt.Printf("collected metrics at %s\n", time.Now().UTC())
```

2. Не используй пакет `fmt`, вместо него используй `log`. В дальнейшем мы перейдем либо на `slog`, либо на сторонний линтер типа `zap`.